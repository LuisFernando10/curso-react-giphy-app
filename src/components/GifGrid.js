    
    import React from 'react';
    import PropTypes from 'prop-types';
    import { useFetchGifs } from '../hooks/useFetchGifs';
    import { GifGridItem } from './GifGridItem';

    export const GifGrid = ( { category } ) => {

        /*const [images, setImages] = useState( [] );

        /// El Hook 'UseEffect', permite ejecutar una única vez cuando el componente es renderizado por primera vez
        useEffect( () => {
            getGifs( category )
                .then( setImages );// Sólo se coloca la referencia al el método porque la promesa y el método reciben un parámetro, entonces se omite
        }, [ category ]);// Agregamos la 'categoría' para que si 'cambia', entonces ejecutamos nuevamente el Effect
        */

        const { data:images, loadig } = useFetchGifs( category ); // 'data:images', es la forma de renombrar una 'prop' cuando en el archivo local, queremos personalizar su nombre

        return (
            <>
                <h2>{ category }</h2>

                {/* { loadig ? <p>Cargando...</p> : null } //Ésta es la forma larga de evaluar dos condiciones  */}
                { loadig && <p className='animate__animated animate__shakeX'>Cargando...</p> }
                <div className='card-grid'>
                    {
                        images.map( ( img ) => (
                            <GifGridItem
                                key={ img.id }
                                { ...img }
                            />
                        ))
                    }
                </div>
            </>
        )
    }

    GifGrid.propTypes = {
        category: PropTypes.string.isRequired
    }