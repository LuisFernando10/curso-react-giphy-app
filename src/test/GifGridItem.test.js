
    import React from 'react';
    import '@testing-library/jest-dom';
    import { shallow } from 'enzyme';
    import { GifGridItem } from '../components/GifGridItem';
    
    describe('Pruebas al componente: GifGridItem.js', () => { 

        const title = 'Imagen Prueba';
        const url = 'https://localhost/algo.jpg';
        const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );
        
        test('Retorna: Componente <GifGridItem />', () => { 
            expect( wrapper ).toMatchSnapshot();
        });

        test('Retorna: String con el párrafo del title', () => { 

            const element_p = wrapper.find('p');
            expect( element_p.text().trim() ).toBe( title );
        });

        test('Retorna: Strings con las props del componente', () => { 

            const element_img = wrapper.find('img');
            
            // Dos formas diferentes de recuperar las 'props' del componente mediante los métodos: 'props() y prop()'
            expect( element_img.props().src ).toBe( url );
            expect( element_img.prop('alt') ).toBe( title );
        });

        test('Valida: Clase animate__fadeInDown', () => { 

            const element_div = wrapper.find('div');
            
            //const clas_name = element_div.prop( 'className' );
            //expect( className.includes('animate__fadeInDown') ).toBe( true );
            
            expect( element_div.hasClass('animate__fadeInDown') ).toBe( true );
        });
     });