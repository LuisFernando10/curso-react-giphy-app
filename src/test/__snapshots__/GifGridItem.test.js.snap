// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente: GifGridItem.js Retorna: Componente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Imagen Prueba"
    src="https://localhost/algo.jpg"
  />
  <p>
    Imagen Prueba
  </p>
</div>
`;
